src/frontend/src/utils/credential-devices.ts

```diff
@@ -13,8 +13,22 @@ const derFromPubkey = (pubkey: DeviceKey): DerEncodedPublicKey =>
 
 export const convertToCredentialData = (
   device: Omit<DeviceData, "alias">
-): CredentialData => ({
-  credentialId: Buffer.from(device.credential_id[0] ?? []),
-  pubkey: derFromPubkey(device.pubkey),
-  origin: device.origin[0],
-});
+): CredentialData | undefined => {
+  // In certain cases, e.g. Chrome on Windows 10, an invalid credential id is
```

Comment:
Why did you move this logic in `convertToCredentialData`? It feels they are different conceptual things.

Reply:
That was intentional, by moving it into the same function it guarantees credential data _always_ has a valid credential id.

I initially experimented with a `hasValidCredentialId` function which with the correct typing can guarantee that the credential id is not undefined in the device. But this didn't give the guarantee that the credential id is valid (more than 0 bytes) when it is set in the device. 

Checking the latter with a type guard would be difficult during TS compile - since we want to avoid runtime errors not create them by throwing an error - hence I opted instead to move the check into the function itself and force the caller to handle the case where a credential doesn't have a valid credential id.

Reply:
My issue is that we might need "valid" credentials somewhere else in the code and in this case we just happen to be using the same converting function in those two places we also need valid credentials.

The same way I didn't realize the flatMap was important filtering out those credentials. I'm afraid the bug might be introduced again in the future. How about renaming the function to `convertToValidCredentialData`?

Reply:
Good idea, I'll rename the function.